znai-reactjs/src/doc-elements/references/globalDocReferences.ts

```diff
@@ -14,21 +15,27 @@
  * limitations under the License.
  */
 
+import { DocReferences } from "./DocReferences";
+
 /**
  * global references are defined on the documentation level and is applicable to every code snippet.
  * references is loaded as a javascript file with a globally defined variable (similar to how Table Of Contents is loaded)
  *
  * @param localReferences local references, can be null
  * @returns {Object} global references merged with local, null if neither is defined
  */
-export function mergeWithGlobalDocReferences(localReferences) {
-    return {...getGlobalDocReferences(), ...localReferences}
+export function mergeWithGlobalDocReferences(
+  localReferences: DocReferences | undefined
```

Comment:
(completely insignificant) nit: `localReferences?: DocReferences` is equivalent

Reply:
I had an impression that it only works on the last params in a function. could be totally confusing this with something else.

Reply:
just tested it, vscode seems happy with it (though I think python has that rule for optional/default params)